### PR TITLE
Fix double disconnects from exceptions caused by client closing (#1604)

### PR DESCRIPTION
**Pull request by Mystiflow**
#1604

### CODE REVIEW

**Comment by kamcio96** [comment]
#1874

**Comment by md-5** [comment]
Given your reasoning, it would be correct to set the server as obsolete inside the con.disconnect method to catch all call cases, not just this specific one.

**Comment by Mystiflow** [comment]
Updated to catch more use cases

File: proxy/src/main/java/net/md_5/bungee/UserConnection.java [hunk 1]
```diff
@@ -140,6 +140,7 @@
     @Getter
     @Setter
     private ForgeServerHandler forgeServerHandler;
+    private boolean disconnecting;
```
Comment (md-5) [inline proxy/src/main/java/net/md_5/bungee/UserConnection.java]: This should be part of ChannelWrapper. You also haven't rationalized its existence.

Comment (Mystiflow) [inline proxy/src/main/java/net/md_5/bungee/UserConnection.java]: The disconnect method currently only closes the channel after 500ms; without this flag, this method can be called multiple times within a 500ms time-frame and still log in the console that the user has disconnected and create a runnable to close the channel and send a kick packet.

By setting this as a dirty flag, we can ignore additional cases.

**Comment by md-5** [comment]
I'm just gonna bundle this with some more generic changes to improve overall code if thats alright.

**Pull request closed without merging**